src/app/pages/MarginTradePage/components/TradingHistory/utils.tsx

```diff
@@ -0,0 +1,337 @@
+  position: TradingPosition;
+  type: EventType;
+  leverage: string;
+  positionSize: string;
+  price: string;
+  txHash: string;
+  isOpen: boolean;
+  time: number;
+  event: string;
+  collateralToLoanRate: string;
+}
+
+export interface IEventData {
+  data: Array<ICustomEvent>;
+  isOpen: boolean;
+  loanId: string;
+}
+
+export interface ICalculatedEvent {
+  loanId: string;
+  position: TradingPosition;
+  loanToken: Asset;
+  collateralToken: Asset;
+  leverage: string;
+  positionSize: string;
+  entryPrice: string;
+  closePrice: string;
+  profit: string;
+  entryTxHash: string;
+  closeTxHash: string;
+  time: number;
+}
+
+export const HistoryTable = (props: { items: ICalculatedEvent[] }) => {
+  const { t } = useTranslation();
+  const [page, setPage] = useState(1);
+  const pageLimit = 5;
+
+  const data = React.useMemo(() => {
+    return props.items
+      .slice(page * pageLimit - pageLimit, page * pageLimit)
+      .map(item => {
+        const pair = TradingPairDictionary.findPair(
+          item.loanToken,
+          item.collateralToken,
+        );
+
+        if (pair === undefined) return null;
+
+        const isLong = pair.longAsset === item.loanToken;
+
+        return {
+          item: item,
+          icon: <PositionBlock position={item.position} name={pair.name} />,
+          leverage: `${item.leverage}x`,
+          positionSize: (
+            <Tooltip content={weiTo18(item.positionSize)}>
+              <span>
+                {weiToNumberFormat(item.positionSize, 4)}{' '}
+                {isLong ? item.loanToken : item.collateralToken}
+              </span>
+            </Tooltip>
+          ),
+          entryPrice: (
+            <>
+              {weiToNumberFormat(item.entryPrice, 4)}{' '}
+              {isLong ? item.loanToken : item.collateralToken}
+            </>
+          ),
+          closePrice: (
+            <>
+              {weiToNumberFormat(item.closePrice, 4)}{' '}
+              {isLong ? item.loanToken : item.collateralToken}
+            </>
+          ),
+          profit: (
+            <TradeProfit
+              positionSize={item.positionSize}
+              pair={pair}
+              loanId={item.loanId}
+              loanToken={item.loanToken}
+              closePrice={item.closePrice}
+              entryPrice={item.entryPrice}
+              position={item.position}
+              asset={item.collateralToken}
+            />
+          ),
+          entryTxHash: (
+            <LinkToExplorer
+              txHash={item.entryTxHash}
+              className="tw-text-primary tw-truncate"
+              startLength={5}
+              endLength={5}
+            />
+          ),
+          closeTxHash: (
+            <LinkToExplorer
+              txHash={item.closeTxHash}
+              className="tw-text-primary tw-truncate"
+              startLength={5}
+              endLength={5}
+            />
+          ),
+        };
+      })
+      .filter(item => !!item);
```

Comment:
What does this filter doing?

Reply:
i think It is superfluous here, maybe it was used with other endpoint for some reasons
anyway, the result is the same without this filter